java/datastore/src/main/java/com/google/datastore/v1/client/RemoteRpc.java

```diff
@@ -110,6 +127,22 @@ public InputStream call(String methodName, MessageLite request) throws Datastore
     }
   }
 
+  @VisibleForTesting
+  void setHeaders(MessageLite request, HttpRequest httpRequest) {
+    httpRequest.getHeaders().put(API_FORMAT_VERSION_HEADER, API_FORMAT_VERSION);
+    if (enableE2EChecksum && request != null) {
+      byte[] checksum = EndToEndChecksumHandler.computeChecksum(request.toByteArray());
```

Comment:
I don't think the bytes returned by digest are guaranteed to be URL safe.
Does the backend expect hex, base62, base64?

Reply:
ah that is a big bug in the API. I changed it to be URLEncode'd String.
This is forcing JDK source/target in pom.xml to be at least 10.
please review if that is acceptable.